File: packages/expo-video/android/src/main/java/expo/modules/video/enums/DRMType.kt
package expo.modules.video.enums

import androidx.media3.common.C
import expo.modules.kotlin.types.Enumerable
import expo.modules.video.UnsupportedDRMTypeException
import java.util.UUID

enum class DRMType(val value: String) : Enumerable {
  CLEARKEY("clearkey"),
  FAIRPLAY("fairplay"),
  PLAYREADY("playready"),
  WIDEVINE("widevine");

  fun isSupported(): Boolean {
    return this != FAIRPLAY
  }

  fun toUUID(): UUID {
    return when (this) {
      CLEARKEY -> C.CLEARKEY_UUID
      FAIRPLAY -> throw UnsupportedDRMTypeException(this)
      PLAYREADY -> C.PLAYREADY_UUID
      WIDEVINE -> C.WIDEVINE_UUID
    }
  }
}

File: packages/expo-video/ios/Enums/DRMType.swift
// Copyright 2024-present 650 Industries. All rights reserved.

import ExpoModulesCore

internal enum DRMType: String, Enumerable {
  case clearkey
  case fairplay
  case playready
  case widevine

  func isSupported() -> Bool {
    return self == .fairplay
  }

  internal func assertIsSupported() throws {
    if !isSupported() {
      throw DRMUnsupportedException(self)
    }
  }
}
